### ISSUE TITLE
bug: Cody keeps reverting code back to previous code

### ISSUE THREAD

**Issue by endolith** [sourcegraph/cody#5324]
### Version

v1.32.2

### Describe the bug

In the last day or so, Cody seems to ignore any context I place in the message box, and defers to previous conversation context instead.  If it has Items [a, b, c] in a list, for instance, and that order has appeared in the conversation previously, and then I manually change it to [c, b, a] and save the file, and then ask Cody to add line d, it will generate code with [a, b, c, d] instead of [c, b, a, d], using the old order rather than building off of the current code that I included in context.  This is with Claude 3.5 Sonnet.

Also at the same time, clicking on the downvote button no longer does anything, it stays hollow black:

![image](https://github.com/user-attachments/assets/533b7239-ff0a-4f37-89a9-43ca151ca13f)

### Expected behavior

If I include a file in context using Alt+L, it should base its modifications on that file as it currently exists, not on the previous conversation.

### Additional context

Did prompts change in the latest update or something that would cause this?

_No response_

**Comment by endolith** [comment]
For example:

![image](https://github.com/user-attachments/assets/eeca9987-f76b-4254-9188-39b2d5e37e17)

Despite me explicitly highlighting the new abbreviated labels and sending them to the chat context:

![image](https://github.com/user-attachments/assets/d4426f28-04f3-400e-9b7d-9dc7895d87cb)

Every time I try, it insists on its own previous output rather than the code I am showing to it, unless I copy and paste it myself:

![image](https://github.com/user-attachments/assets/07f704b4-2d38-43d7-86e7-6916c30ae34a)

**Comment by endolith** [comment]
I highlight this line and try again:

![image](https://github.com/user-attachments/assets/8a8c8881-e9b2-4e58-b91b-b6576381c91d)

but it gives a different line:

![image](https://github.com/user-attachments/assets/a58daaca-e50e-4311-a24a-e106bc606163)

I highlight this line and try again:

![image](https://github.com/user-attachments/assets/fff9a554-bc06-4dce-b593-8021800d82e1)

But it gives a line from a completely different file:

![image](https://github.com/user-attachments/assets/c4ac5068-35ba-42fb-bedb-78e646056932)

If I start a new chat, it says it doesn't see anything at all, which is likely the underlying problem?

![image](https://github.com/user-attachments/assets/a21d3c80-22b0-4d45-98c7-a96a57fedf35)

**Comment by endolith** [comment]
I reverted to v1.30.3 but it's still happening.  😣

**Comment by chenkc805** [comment]
Thanks @endolith for flagging this bug. I believe this is either than indexing bug with our local context fetching or just a plan bug on our end. Will address

**Comment by endolith** [comment]
It may just be an issue with that particular conversation?  Maybe because it's very long?

**Comment by chenkc805** [comment]
Yeah I don't think it's a length issue, I've been able to repro this bug on my own end with a short conversation

**Comment by derjogi** [comment]
It seems like once cody has context, it doesn't get updated any more, but if you give it new context it works fine. 

E.g. starting a new chat with context always works, and if I start a chat and give it only A.tsx, and then later B.tsx as context it has accurate knowledge of B.tsx.
However, If I give it directly A.tsx & B.tsx, then later change something in either and provide it again with the files, it doesn't have the updated knowledge.

I haven't structurally tried to proof that this happens all the time, but it feels like it does.

**Comment by duskosusa** [comment]
Yeah, context doesn't seem to change mid convo, even if I delete embeddings folder and it rebuilds it

**Comment by endolith** [comment]
This is still a frequent problem :/

**Comment by endolith** [comment]
Still happening.  I am going to try your competitors now.  :(

![image](https://github.com/user-attachments/assets/cd89348e-b654-452f-ba6a-2ee6f7a4abda)

**Comment by davidbien** [comment]
Yeah this apparently isn't just a bug, but is how things work. If you want Cody to use the newest code you MUST paste the latest code wholesale into the chat window. Otherwise it will used cached code that you may have changed 20 times on the screen. This is a horrible problem. The way around it, as far as I can tell, is to open a new conversation.
This bug really really really needs fixing - it will reintroduce bugs you have fixed and you may not even realize that Cody changed the code back to before the bug you fixed in the background.
I have less serious complaints about this product, but this is by far the most serious issue in this product that I have experienced at this moment.

**Comment by endolith** [comment]
@davidbien 

I switched to Cursor... 💀

It's more expensive though, so Cody could still win me back by fixing things, maybe. 😬

**Comment by davidbien** [comment]
Cursor doesn't integrate in vscode with a plugin - this is key cuz I'm already using Xcode and vscode to develop my iOS app, and I just think that Cursor's decision to supply an editor is so weak that I didn't even try it - like I want to be dependent on *another* editor.

**Comment by endolith** [comment]
Cursor *is* VScode, though, so it runs all the same plugins, but with less clunky integration of the AI.

**Comment by davidbien** [comment]
Yeah I tried Cody for a bit. It lost the plot on the chat we were having after like 2-3 prompts - suddenly it wondered what we were talking about - what problem I was having. Cody works *tons* better than that - like night and day. It might be a problem that it maintains cached code - but it isn't a problem that it maintains an incredible chat context that you can even go back hours later and ask it to restore the comments on a method it generated where it stomped on my comments and it did it perfectly. That's pretty sick.
One thing is though: *even when* I post the new code straight into the chat window it may *even still* use a cached version of that code when generating. I have to then tell it to please use the code I just pasted and it will. The persistence of old code from long ago in the conversation is impressive honestly - but also sucks as we understand.

**Comment by davidbien** [comment]
The key, really, when working with Cody's impl of Claude 3.5 Sonnet, is: if you are working on a class in the chat window - which is my primary modus operandi, make any changes to the code using Cody. Even if you want to make a small change to the class that the AI didn't notice - which happens all the time - use the chat window to make that change. This keeps the AI's idea of the code in its context the same as what is in the editor. If you go and fix a problem on your own and don't tell Cody you are going to do it then Claude's context diverges from what is in the actual source file.

**Comment by endolith** [comment]
Yeah, I've seen that happen in both.  (This bug is a different problem, though, where the AI is literally being fed the wrong lines of code.)
